### ISSUE TITLE
pts/compress-zstd-1.2.1 is using a LOT of memory on ARM64

### ISSUE THREAD

**Issue by MarcDoughty** [phoronix-test-suite/test-profiles#183]
I noticed that my 4GB Raspberry Pi 4 was going deep into swap on the pts/compress-zstd-1.2.1 tests, leading to results that are far below expected.

Running Ubuntu Groovy ARM64.

**Comment by michaellarabel** [comment]
It might just not be a good test to run then on the Raspberry Pi 4GB. Or I am not sure what the issue report is for? PTS is just using the upstream Zstandard implementation with a real-world Ubuntu ISO as an example input and in the multi-threaded mode but no other options or modifications outside of that. If the file is too small, it's not relevant for testing of modern, higher-end hardware. Or am I missing out on something?

**Comment by MarcDoughty** [comment]
Yeah, I was mulling over whether it made sense to file this or not, but the memory difference between 'system/compress-zstd' and 'pts/compress-zstd' brought me to your doorstep.

It looks like the system zstd tops-out at about 280MB memory usage during the test, while the one built by PTS is using gigabytes, perhaps more than logically makes sense while compressing an 800MB file. I was thinking that compiling zstd with -O1 -O2 or -Os might tame it.

I have no idea what the implications for the test results would be, just wanted to raise this as a potential problem. Seems like it be more of an upstream issue.
